build.gradle

```diff
@@ -1535,31 +1573,24 @@ void addMavenPublication(Project project, boolean hasNative, List<String> projec
 
                     if (!projectDependencies.empty) {
                         pom.withXml {
+                            Node parent = asNode().appendNode("parent")
+                            parent.appendNode("groupId", MAVEN_GROUP_ID)
+                            parent.appendNode("artifactId", "javafx")
+                            parent.appendNode("version", MAVEN_VERSION)
+
                             Node dependencies = asNode().appendNode("dependencies")
                             projectDependencies.each { dep ->
                                 Node dependency = dependencies.appendNode("dependency")
                                 dependency.appendNode("groupId", MAVEN_GROUP_ID)
                                 dependency.appendNode("artifactId", "javafx.$dep")
                                 dependency.appendNode("version", MAVEN_VERSION)
-                                dependency.appendNode("classifier", t.name)
+                                dependency.appendNode("classifier", "\${javafx.platform}")
```

Comment:
I'm pretty sure we can access the "os.name" and "os.arch" system properties directly from Gradle without Maven...

Reply:
Yeah, but Maven doesn't give you any sort of control over those values, so it's pretty limiting. I'll eventually be changing that in ND4J as well. Here's what is look like in JavaCPP:
https://github.com/bytedeco/javacpp/blob/master/src/main/java/org/bytedeco/javacpp/Loader.java#L83-L112

Reply:
We have the additional constraint that javafx.graphics should be a single module, hence a single jar. That module contains the NativeLibLoader, so that is part of the module (but exists in different flavors, one per module).
I hope that in the not-too-far future maven and gradle have better support for platform specific classifiers *and* modules.
So far, the current solution seems to work (tested on mac/linux/windows). 
A cool feature of nd4j/javacpp which we don't have yet is the possibility of cross-compilation (e.g. you can specify javacpp.platform=ios on a mac). Would be great to have this later as well.

For now, the main goal is to make sure we can provide the module functionality with artifacts uploaded to maven central so that developers can leverage module-info statements in their projects.
That seems to work now.

Reply:
I'm pretty sure that the dependency doesn't work from anything else than Maven though, only Maven supports profiles.

Reply:
probably. But I think the build process should make sure the correct code (Java classes and native libs) are selected, instead of deciding this at runtime. With the move towards AOT this sounds plausible. But it's an interesting discussion point.

Reply:
It will always need to happen at runtime, we're talking WORA after all, unless we don't and go the JMOD/app way only, in which case it happens at build time for each end-user app.

Reply:
I don't see the link with WORA. WORA is about writing your *code* once. Hence, putting build/dependency logic in build systems (which is more or less their reason of existence) doesn't violate WORA.

Reply:
We frequently end up repackaging artifacts in a "uber JAR", and if that JAR file contains only binaries for Mac because the developer is on Mac, it's not going to run on Linux.

Anyway, there's lots of other corner cases as well. You're far from being out of the woods... I do hope people eventually understand the value of standardizing all that in something like JavaCPP.